dev/com.ibm.ws.security.spnego_fat/fat/src/com/ibm/ws/security/spnego/fat/FATSuite.java

```diff
@@ -0,0 +1,169 @@
+
+    @ClassRule
+    public static RepeatTests repeat = RepeatTests.withoutModification().andWith(new JakartaEE9Action());
+
+    @ClassRule
+    public static ExternalResource testRule = new ExternalResource() {
+        /**
+         * Creates the SPN and keytab file to be used in any ensuing tests. Test classes can elect to create their own
+         * SPN and keytab file if needed. A common SPNEGO token is also created which can be used by all test classes.
+         * The common SPNEGO token should be refreshed at a certain interval, specified by the
+         * TOKEN_REFRESH_LIFETIME_SECONDS variable.
+         */
+        @Override
+        protected void before() throws Exception {
+            String thisMethod = "before";
+            Log.info(c, thisMethod, "Performing the common setup for all test classes");
+
+            getKDCInfoFromConsul();
+
+            if (!isSupportJDK())
+                return;
+
+            String ip = InetAddress.getByName("localhost").getHostAddress();
+            if (!"127.0.0.1".equals(ip)) {
+                Log.info(c, thisMethod, "The localhost ip address is " + ip + " (not default value 127.0.0.1), localhost ip address tests will not be run");
+                LOCALHOST_DEFAULT_IP_ADDRESS = false;
+            }
+
+            try {
+                CommonTest.commonSetUp("com.ibm.ws.security.spnego.fat.setup", null, SPNEGOConstants.NO_APPS, SPNEGOConstants.NO_PROPS,
+                                       SPNEGOConstants.DONT_CREATE_SSL_CLIENT,
+                                       SPNEGOConstants.CREATE_SPN_AND_KEYTAB, SPNEGOConstants.DEFAULT_REALM, SPNEGOConstants.CREATE_SPNEGO_TOKEN,
+                                       SPNEGOConstants.SET_AS_COMMON_TOKEN, SPNEGOConstants.USE_CANONICAL_NAME, SPNEGOConstants.DONT_USE_COMMON_KEYTAB,
+                                       SPNEGOConstants.DONT_START_SERVER);
+            } catch (Exception e) {
+                String message = CommonTest.maskHostnameAndPassword(e.getMessage());
+                Log.info(c, thisMethod, "Common setup failed, tests will not run: " + message, e);
+                throw (new Exception("Common setup failed, tests will not run: " + message, e));
+            }
+
+            // The keytab file needs to be copied into the /tmp folder so we can include it in the other servers
+            CommonTest.getMyServer().copyFileToTempDir(SPNEGOConstants.KRB_RESOURCE_LOCATION.substring(1) + SPNEGOConstants.KRB5_KEYTAB_FILE, SPNEGOConstants.KRB5_KEYTAB_FILE);
+            KEYTAB_FILE_LOCATION = "tmp/" + SPNEGOConstants.KRB5_KEYTAB_FILE;
+
+            Log.info(c, thisMethod, "Common setup is complete");
+        }
+
+        private boolean isSupportJDK() throws IOException {
+            String thisMethod = "isSupportJDK";
+            JavaInfo javaInfo = JavaInfo.forServer(LibertyServerFactory.getLibertyServer("BasicAuthTest"));
+
+            JDK_V7_LOWER = javaInfo.majorVersion() <= 7;
```

Comment:
No need for this code since we don't support Java 7 any longer.

Reply:
I will remove this.

Reply:
I got rid of the `JDK_V7_LOWER` field in InitClass, which meant I could remove the `testS4U2SelfJDKVersionComplianceTest` test in `S4U2SelfTest.java`.

Reply:
I also removed this from the two other FATSuite classes as well.